src/test/scala/com/fulcrumgenomics/fgsv/util/PutativeBreakpointTest.scala

```diff
@@ -0,0 +1,98 @@
+  }
+
+  it should "correctly capture an R-R breakpoint" in {
+    // R-R breakpoint if observed with a "forward" read where position in read is correlated to position in genome
+    val block1 = block(readStart=1,  refIndex=1, refStart=100,  length=50, positive=false)
+    val block2 = block(readStart=51, refIndex=1, refStart=1000, length=30, positive=false)
+    val bp     = PutativeBreakpoint(block1, block2, EvidenceType.SplitReadIntraContig)
+    (bp.leftPos, bp.leftPositive)   shouldBe (100, false)
+    (bp.rightPos, bp.rightPositive) shouldBe (1029, false)
+
+    // Same breakpoint observed with a "reverse" read where position in read is anti-correlated to position in genome
+    val revBlock1 = block(readStart=1, refIndex=1, refStart=1000, length=30, positive=true)
+    val revBlock2 = block(readStart=31,  refIndex=1, refStart=100,  length=50, positive=true)
+    val rbp       = PutativeBreakpoint(revBlock1, revBlock2, EvidenceType.SplitReadIntraContig)
+    (rbp.leftPos, rbp.leftPositive)   shouldBe (100, false)
+    (rbp.rightPos, rbp.rightPositive) shouldBe (1029, false)
+
+  }
+
+  it should "correctly capture an R-F breakpoint" in {
+    // R-F breakpoint if observed with a "forward" read where position in read is correlated to position in genome
+    val block1 = block(readStart=1,  refIndex=1, refStart=100,  length=50, positive=false)
+    val block2 = block(readStart=51, refIndex=1, refStart=1000, length=30, positive=true)
+    val bp     = PutativeBreakpoint(block1, block2, EvidenceType.SplitReadIntraContig)
+    (bp.leftPos, bp.leftPositive)   shouldBe (100, false)
+    (bp.rightPos, bp.rightPositive) shouldBe (1000, true)
+
+    // Same breakpoint observed with a "reverse" read where position in read is anti-correlated to position in genome
+    val revBlock1 = block(readStart=1, refIndex=1, refStart=1000, length=30, positive=false)
+    val revBlock2 = block(readStart=31,  refIndex=1, refStart=100,  length=50, positive=true)
+    val rbp       = PutativeBreakpoint(revBlock1, revBlock2, EvidenceType.SplitReadIntraContig)
+    (rbp.leftPos, rbp.leftPositive)   shouldBe (100, false)
+    (rbp.rightPos, rbp.rightPositive) shouldBe (1000, true)
+  }
+
+  it should "represent a tandem duplication sanely" in {
+    // We want the same representation regardless of which piece of the duplication was sequenced more so
+    // we test where the second block is shorter, the same length as, and longer than the first
+    Seq(-1, 0, 1).foreach { addend =>
+      val block1Length = 50
+      val block2Length = block1Length + addend
+      val block1 = block(readStart=1,  refIndex=1, refStart=100, length=block1Length, positive=true)
+      val block2 = block(readStart=51, refIndex=1, refStart=100, length=block2Length, positive=true)
+      val bp     = PutativeBreakpoint(block1, block2, EvidenceType.SplitReadIntraContig)
+      (bp.leftPos,  bp.leftPositive)  shouldBe (100, false)
+      (bp.rightPos, bp.rightPositive) shouldBe (149, false)
```

Comment:
So the test case here is a tandem duplication of 50bp at positions 100-149.  So we basically read that chunk of genome twice sequentially, and make a breakend out of it.

Personally I would find it more intuitive to have the breakend information laid out so that it says:
- You're going along chr1 and get to base 149, and then
- Suddenly you find yourself back at base 100 again, still reading in the same direction

I.e. the left hand side would be `149+` and the right hand side would be `100+`, but given our existing definition of canonical as "left hand side lower" we use up with the left hand side being `100-` and the right hand being `149-`.  I _think_ if I had my druthers I would canonicalize so that the left hand side was always positive even if that meant having it be higher on the genome, but I haven't thought about it hard enough to know if that's a bad idea or not.